docs/proposals/08-peering.md

```diff
@@ -0,0 +1,168 @@
+---
+title: VNet Peering
+
+oep-number: 8
+
+creation-date: 2023-02-24
+
+status: implementable
+
+authors:
+
+- "@ManuStoessel"
+- "@gehoern"
+
+reviewers:
+
+- "@afritzler"
+- "@adracus"
+
+---
+
+# OEP-NNNN: Your short, descriptive title
+
+## Table of Contents
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+- [Alternatives](#alternatives)
+
+## Summary
+
+Network peering is a crucial feature to allow users to route traffic between two onmetal networks. This OEP introduces a new API that allows the owner of both networks to express the consent to peering the networks explicitely and extends the existing `Network` API to reflect an active peering.
+
+- peering connects whole networks together 
+(vf get both network route set announced)
+- clear separation from transit gateways
+- we need peering requests on both partners
+- peering request should have a ttl so they get removed if not matched
+
+## Motivation
+
+Without peering, users of onmetal would need to run all services via public IPs or via complicated VPN solutions if they are located in seperate networks.
+We want to implement the basic use case of allowing services running in two distinct onmetal networks to be able to reach eachother without any additional installation from the user and without exposing said services to the public internet.
+
+### Goals
+
+* Define API for requesting and acknowledging peering of two network
+* Peering request should have a TTL that allows for garbage collection of unanswered requests
+* Peering should result in one network's prefixes should be fully routeable from the other network's prefixes and vice versa (vf get both network route sets announced)
+
+### Non-Goals
+
+* We do not want to implement a transit gateway
+
+## Proposal
+
+We propose using a new API called `NetworkPeeringRequests` and extending the `Network` API with a `status` reflecting the peering. For a peering to succeed, both owners of the two `Networks` that should peer, have to create a matching `NetworkPeeringRequest`. On successful peering, the `status` of the corresponding `Network` objects should be updated with the other `Network`s details. A `NetworkPeeringRequest` will also have a set time-to-live (ttl), so the system will be able to automatically clean up vacant or unmatched `NetworkPeeringRequest`objects. Always two matching `NetworkPeeringRequests` are required to enable peering. Important: `localNetworkRef` must match to other `remoteNetworkRef` and vice versa.
```

Comment:
The type should be called `NetworkPeering` as it represents the peering. The `Request` suffix does not make sense here.